fearless/Common/Network/Subscan/SubscanHistoryOperationFactory.swift

```diff
@@ -0,0 +1,236 @@
+                }
+                return array
+            }()
+
+            let truncationLength: Int? = arrayOfIndexes
+                .sorted { $0.1 < $1.1 }
+                .first { !(completionMapping[$0.0] ?? false) }
+                .map { $0.1 + 1 }
+
+            let truncatedItems: [WalletRemoteHistoryItemProtocol] = {
+                if let length = truncationLength {
+                    return Array(resultItems.prefix(length))
+                } else {
+                    return resultItems
+                }
+            }()
+
+            return MergeResult(items: truncatedItems, originalCounters: originalCounters)
+        }
+    }
+
+    private func createMapOperation(
+        dependingOn mergeOperation: BaseOperation<MergeResult>,
+        context: TransactionHistoryContext,
+        filter: WalletRemoteHistoryFiltering?
+    ) -> BaseOperation<WalletRemoteHistoryData> {
+        ClosureOperation<WalletRemoteHistoryData> {
+            let mergeResult = try mergeOperation.extractNoCancellableResultData()
+            let counters = mergeResult.items
+                .reduce(into: [WalletRemoteHistorySourceLabel: Int]()) { result, item in
+                    result[item.label] = (result[item.label] ?? 0) + 1
+                }
+
+            let nextContext = WalletRemoteHistorySourceLabel.allCases.reduce(context) { result, label in
+                let sourceContext = result.sourceContext(for: label)
+
+                guard !sourceContext.isComplete else {
+                    return result
+                }
+
+                let mergedCount = counters[label] ?? 0
+                let total = (sourceContext.page * sourceContext.row) + mergedCount
+                let row = total.firstDivider(from: (1 ... result.defaultRow).reversed()) ?? 1
+                let nextPage = total / row
+
+                let originalCount = mergeResult.originalCounters[label] ?? 0
+                let isCompleted = originalCount == mergedCount ? originalCount < sourceContext.row : false
+
+                let nextSourceContext = result.sourceContext(for: label)
+                    .byReplacingPage(nextPage)
+                    .byReplacingRow(row)
+                    .byReplacingCompletion(isCompleted)
+
+                return result.byReplacingSource(context: nextSourceContext, for: label)
+            }
+
+            let finalItems: [WalletRemoteHistoryItemProtocol]
+
+            if let filter = filter {
+                finalItems = mergeResult.items.filter { filter.includes(item: $0) }
+            } else {
+                finalItems = mergeResult.items
+            }
+
+            return WalletRemoteHistoryData(
+                historyItems: finalItems,
+                context: nextContext
+            )
+        }
+    }
+}
+
+extension SubscanHistoryOperationFactory: WalletRemoteHistoryFactoryProtocol {
+    func createOperationWrapper(for context: TransactionHistoryContext, address: String, count _: Int)
```

Comment:
`count` parameter is not used

Reply:
Its part of the protocol and is used by another implementation